File: data/src/main/java/com/smb/data/models/Chapter.java
package com.smb.data.models;

import android.os.Parcel;
import android.os.Parcelable;

public class Chapter implements Parcelable{
    String title;
    Media media;

    public Chapter(String title, Media media) {
        this.title = title;
        this.media = media;
    }

    protected Chapter(Parcel in) {
        title = in.readString();
        media = in.readParcelable(Media.class.getClassLoader());
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(title);
        dest.writeParcelable(media, flags);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public static final Creator<Chapter> CREATOR = new Creator<Chapter>() {
        @Override
        public Chapter createFromParcel(Parcel in) {
            return new Chapter(in);
        }

        @Override
        public Chapter[] newArray(int size) {
            return new Chapter[size];
        }
    };

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public Media getMedia() {
        return media;
    }

    public void setMedia(Media media) {
        this.media = media;
    }
}

File: presentation-tv/src/main/java/com/tv/models/Chapter.kt
package com.tv.models

import android.os.Parcel
import android.os.Parcelable

class Chapter() : Parcelable {
    var title: String? = null
    var media: Media? = null

    constructor(parcel: Parcel) : this() {
        title = parcel.readString()
        media = parcel.readParcelable(Media::class.java.classLoader)
    }

    constructor(title: String, media: Media) : this() {
        this.title = title
        this.media = media
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(title)
        parcel.writeParcelable(media, flags)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Chapter> {
        override fun createFromParcel(parcel: Parcel): Chapter {
            return Chapter(parcel)
        }

        override fun newArray(size: Int): Array<Chapter?> {
            return arrayOfNulls(size)
        }
    }
}
